statsmodels/discrete/discrete_model.py

```diff
@@ -1571,6 +1597,32 @@ def score(self, params):
         else:
             return score
 
+    def score_factor(self, params):
+        if self._transparams:
+            alpha = np.exp(params[-1])
+        else:
+            alpha = params[-1]
+
+        params = params[:-1]
+        p = self.parameterization
+        exog = self.exog
+        y = self.endog[:,None]
+        mu = self.predict(params)[:,None]
+        mu_p = np.power(mu, p)
+        a1 = 1 + alpha * mu_p
+        a2 = mu + alpha * mu_p * y
+        a3 = alpha * p * mu ** (p - 1)
+        a4 = a3 * y
+        dmudb = mu
+
+        dalpha = (mu_p * (y * ((y - 1) / a2 - 2 / a1) + a2 / a1**2))
+        dparams = dmudb * (-a4 / a1 +
+                           a3 * a2 / (a1 ** 2) +
+                           (1 + a4) * ((y - 1) / a2 - 1 / a1) +
+                           1 / mu)
```

Comment:
- [ ] `1/mu` term factors out and when multiplied by `dmudb` is just 1
- [ ] Substituting in `a4=a3*y` I _think_ will make it easier to identify cancellable terms, as well as terms common between this and `dalpha`

Reply:
Musing: I'm confident Mathematica or something has a maximally-simplify command, but that isn't what we want here.  We want some convex combination of maximally-simplify and maximal-readability/interprebility.  I wonder if there is a tool for this.